### PR TITLE
Moved `Help & Support` to Profile screen

### PR DESCRIPTION
**Pull request by yaelirub**
part of  #1234 

Please include a summary of the changes and the related issue.

## To test
1. go to Profile page and see `Help & Support` 
2. Tap on `Help & Support` and see that you get the support web view 
3. Got to Settings and see `Help & Support`  is removed.

https://github.com/Automattic/pocket-casts-ios/assets/1335657/fcf18314-6f1d-4f56-8899-2592d6b60681

Steps to test your PR.

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/ProfileViewController.swift [hunk 1]
```diff
@@ -325,7 +331,7 @@ class ProfileViewController: PCViewController, UITableViewDataSource, UITableVie
 
     private func tableData() -> [[ProfileViewController.TableRow]] {
         var data: [[ProfileViewController.TableRow]]
-        data = [[.allStats, .downloaded, .uploadedFiles, .starred, .listeningHistory]]
+        data = [[.allStats, .downloaded, .uploadedFiles, .starred, .listeningHistory, .help]]
```
Comment (leandroalonso) [inline podcasts/ProfileViewController.swift:334]: What do you think about having the help cell in its own section?

```suggestion
        data = [[.allStats, .downloaded, .uploadedFiles, .starred, .listeningHistory], [.help]]
```

https://github.com/Automattic/pocket-casts-ios/assets/7040243/17b9b63d-d4bd-4fef-a9db-d3cd6b4ce477

I had to make a few other changes for this to work, documenting them here:

```swift
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        if EndOfYear.isEligible && indexPath.section == 0 && indexPath.row == 0 {
            return UITableView.automaticDimension
        } else {
            return 70
        }
    }

    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
        if section == 0 {
            return headerView
        }

        return nil
    }

    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        if section == 0 {
            return headerViewModel.contentSize?.height ?? UITableView.automaticDimension
        }

        return UITableView.automaticDimension
    }
```

**Review by leandroalonso — COMMENTED** [review]
@yaelirub works as described! Left one comment about having the Help in it's own section, let me know what do you think!

**Comment by yaelirub** [comment]
Hey @leandroalonso, thanks so much for reviewing! I agree that `Help & Feedback` might not feel very connected to the other items on the list, but having it in its own section seems a bit weird to me. Will we have other items in that section?

**Comment by leandroalonso** [comment]
> Will we have other items in that section?

At the moment I don't think so. Anyways, this is a small thing so we're good to go!

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**